src/jit/block.cpp

```diff
@@ -595,9 +596,14 @@ void BasicBlock::CloneBlockState(Compiler* compiler, BasicBlock* to, const Basic
 
     for (GenTreePtr fromStmt = from->bbTreeList; fromStmt != nullptr; fromStmt = fromStmt->gtNext)
     {
-        compiler->fgInsertStmtAtEnd(to,
-                                    compiler->fgNewStmtFromTree(compiler->gtCloneExpr(fromStmt->gtStmt.gtStmtExpr)));
+        auto newExpr = compiler->gtCloneExpr(fromStmt->gtStmt.gtStmtExpr, 0, varNum, varVal);
+        if (!newExpr)
+        {
+            return false;
+        }
```

Comment:
Can you add a comment saying that if gtCloneExpr fails here (because it is too complex, etc...) we return false
(And I would guess abandon any expressions that we have already cloned?)